app/src/main/java/org/simple/clinic/home/overdue/removepatient/RemoveAppointmentSheet.kt

```diff
@@ -0,0 +1,86 @@
+import org.simple.clinic.widgets.UiEvent
+import java.util.UUID
+import javax.inject.Inject
+
+class RemoveAppointmentSheet : BottomSheetActivity() {
+
+  companion object {
+    private const val KEY_APPOINTMENT_UUID = "KEY_APPOINTMENT_UUID"
+
+    fun intent(context: Context, appointmentUuid: UUID) =
+        Intent(context, RemoveAppointmentSheet::class.java)
+            .putExtra(RemoveAppointmentSheet.KEY_APPOINTMENT_UUID, appointmentUuid)!!
+  }
+
+  @Inject
+  lateinit var controller: RemoveAppointmentSheetController
+
+  private val alreadyVisitedRadioButton by bindView<RadioButton>(R.id.removeappointment_reason_patient_already_visited)
+  private val movedOutRadioButton by bindView<RadioButton>(R.id.removeappointment_reason_patient_moved)
+  private val notRespondingRadioButton by bindView<RadioButton>(R.id.removeappointment_reason_patient_not_responding)
+  private val diedRadioButton by bindView<RadioButton>(R.id.removeappointment_reason_patient_died)
+  private val otherReasonRadioButton by bindView<RadioButton>(R.id.removeappointment_reason_other)
+  private val reasonSelectedDoneButton by bindView<PrimarySolidButton>(R.id.removeappointment_done_button)
+
+  private val onDestroys = PublishSubject.create<Any>()
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+
+    setContentView(R.layout.sheet_remove_appointment)
+    TheActivity.component.inject(this)
+
+    Observable.merge(sheetCreates(), alreadyVisitedClicks(), cancelReasonClicks(), doneClicks())
+        .observeOn(io())
+        .compose(controller)
+        .observeOn(mainThread())
+        .takeUntil(onDestroys)
+        .subscribe { uiChange -> uiChange(this) }
+  }
+
+  private fun sheetCreates(): Observable<UiEvent> {
+    val appointmentUuid = intent.extras.getSerializable(KEY_APPOINTMENT_UUID) as UUID
+    return Observable.just(RemoveAppointmentSheetCreated(appointmentUuid))
+  }
+
+  private fun doneClicks(): Observable<UiEvent> = RxView.clicks(reasonSelectedDoneButton).map { RemoveReasonDoneClicked }
+
+  private fun alreadyVisitedClicks() = RxView.clicks(alreadyVisitedRadioButton).map { AlreadyVisitedReasonClicked }
+
+  private fun cancelReasonClicks(): Observable<UiEvent> {
+    val movedOutStream = RxView.clicks(movedOutRadioButton).map { CancelReasonClicked(Appointment.CancelReason.MOVED) }
+    val notRespondingStream = RxView.clicks(notRespondingRadioButton).map { CancelReasonClicked(Appointment.CancelReason.PATIENT_NOT_RESPONDING) }
+    val deathStream = RxView.clicks(diedRadioButton).map { CancelReasonClicked(Appointment.CancelReason.DEAD) }
+    val otherStream = RxView.clicks(otherReasonRadioButton).map { CancelReasonClicked(Appointment.CancelReason.OTHER) }
+
+    return Observable.merge(movedOutStream, notRespondingStream, deathStream, otherStream)
+  }
+
+  fun closeSheet() = finish()
+
+  fun enableDoneButton() {
+    reasonSelectedDoneButton.isEnabled = true
```

Comment:
I just realized we don't have a selector for showing a grey color when this button is disabled. This problem also exists in medical history, so I'll do it as part of my PR.

Reply:
Vinay is working on this